gmso/formats/mol2.py

```diff
@@ -0,0 +1,177 @@
+"""Convert to and from a TRIPOS mol2 file."""
+# TODO add sources of mol2 files
+import warnings
+from pathlib import Path
+
+import unyt as u
+
+from gmso import Atom, Bond, Box, Topology
+from gmso.core.element import element_by_name, element_by_symbol
+from gmso.formats.formats_registry import loads_as
+
+
+@loads_as(".mol2")
+def from_mol2(filename, site_type="atom"):
+    """Read in a TRIPOS mol2 file format into a gmso topology object.
+
+    Creates a Topology from a mol2 file structure. This will read in the
+    topological structure (sites, bonds, and box) information into gmso.
+    Note that parameterized information can be found in these objects, but
+    will not be converted to the Topology.
+
+    Parameters
+    ----------
+    filename : string
+        path to the file where the mol2 file is stored.
+    site_type : string ('atom' or 'lj'), default='atom'
+        tells the reader to consider the elements saved in the mol2 file, and
+        if the type is 'lj', to not try to identify the element of the site,
+        instead saving the site name.
+
+    Returns
+    -------
+    top : gmso.Topology
+
+    Notes
+    -----
+    It may be common to want to create an mBuild compound from a mol2 file. This is possible
+    by installing [mBuild](https://mbuild.mosdef.org/en/stable/index.html)
+    and converting using the following python code:
+
+        >>> import gmso.Topology
+        >>> from gmso.external.convert_mbuild import to_mbuild
+        >>> top = Topology.load('myfile.mol2')
+        >>> mbuild_compound = to_mbuild(top)
+    """
+    msg = "Provided path to file that does not exist"
+    path = Path(filename)
+    if not path.exists():
+        raise OSError(msg)
+    # Initialize topology
+    topology = Topology(name=path.stem)
+    # save the name from the filename
+    f = open(path, "r")
+    line = f.readline()
+    while f:
+        # check for header character in line
+        if line.startswith("@<TRIPOS>"):
+            # if header character in line, send to a function that will direct it properly
+            line = parse_record_type_indicator(f, line, topology, site_type)
+        elif line == "":
+            break
+        else:
+            # else, skip to next line
+            line = f.readline()
+    f.close()
+    topology.update_topology()
+    # TODO: read in parameters to correct attribute as well. This can be saved in various rti sections.
+    return topology
+
+
+def load_top_sites(f, topology, site_type="atom"):
+    """Take a mol2 file section with the heading '<TRIPOS>ATOM' and save to the topology.sites attribute."""
```

Comment:
Can you add a full docstring here?

Just so we can enumerate the possible values for "site_type" etc.